### PR TITLE
Fuzzer: Fix wasm2c name mangling

### PR DESCRIPTION
**Pull request by kripken**
This is necessary for cases where the input has an export name that
needs mangling, like a name with `-` (common in the spec test suite).

### CODE REVIEW

File: src/tools/wasm2c-wrapper.h [hunk 1]
```diff
@@ -25,6 +25,27 @@
...
+      mangled += c;
+    } else {
+      // This must be escaped, as prefix + hex character code.
+      mangled += escapePrefix;
+      std::stringstream ss;
+      ss << std::hex << std::uppercase << unsigned(c);
+      mangled += ss.str();
+    }
+  }
+  return std::string("(*Z_") + mangled + "Z_";
```
Comment (aheejin) [inline src/tools/wasm2c-wrapper.h:46]: I compared this code with wasm2c's [`MangleName`](https://github.com/WebAssembly/wabt/blob/1eb5e2c5b7409a64c6e610fb74df3e5b61394cfc/src/c-writer.cc#L408-L425) method and

Comment (kripken) [reply src/tools/wasm2c-wrapper.h]: Thanks, I think you're right, not sure how this worked in my local testing... fixed.

Comment (kripken) [reply src/tools/wasm2c-wrapper.h]: Oh, actually the code worked before, but was confusing, sorry about that. I refactored it now so it's hopefully clearer.

What was unclear was that the "mangle" function only mangled part of the name, leaving the params etc. for later. There is another `Z_` before them. I moved all the mangling logic into a single place now

**Review by binji — APPROVED** [review]
Generally looks right to me

**Review by aheejin — APPROVED** [review]
Sorry, I was confused ;)

**Pull request merged**